library/pem.c

```diff
@@ -130,6 +142,120 @@ static int pem_pbkdf1( unsigned char *key, size_t keylen,
 
     return( ret );
 }
+#else
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+static int pem_pbkdf1( unsigned char *key, size_t keylen,
+                       unsigned char *iv,
+                       const unsigned char *pwd, size_t pwdlen )
+{
+    unsigned char md5sum[16];
+    psa_hash_operation_t operation = PSA_HASH_OPERATION_INIT;
+    size_t output_length = 0;
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+
+    status = psa_hash_setup( &operation, PSA_ALG_MD5 );
+    if( status != PSA_SUCCESS )
```

Comment:
I think that this whole function would benefit from the same, shorter syntax as above:
```suggestion
if( ( status = psa_hash_setup( &operation, PSA_ALG_MD5 ) != PSA_SUCCESS )
    goto exit;
```
and then at exit the conversion between status and ret.

Reply:
I just thought that they are more comapct and will look the same as the non-PSA counterpart.